### ISSUE TITLE
CentOS 6.10: No rule to make target `coverity-scan.sh', needed by `all-am'.  Stop

### ISSUE THREAD

**Issue by 01CGAT** [netdata/netdata#8080]
##### Bug report summary
When installing with "bash <(curl -Ss https://my-netdata.io/kickstart.sh) --dont-wait" I get:

```
make[2]: Entering directory `/tmp/netdata-kickstart-TcKQW2/netdata-v1.19.0-503-g13d7fc4e'
  CC     libnetdata/os.o
  CC     web/api/formatters/json_wrapper.o
  CC     web/api/formatters/charts2json.o
  CC     web/api/formatters/rrdset2json.o
  CC     exporting/read_config.o
  CC     exporting/init_connectors.o
  CC     exporting/process_data.o
  CC     exporting/check_filters.o
  CC     exporting/send_data.o
  CC     exporting/send_internal_metrics.o
make[2]: *** No rule to make target `coverity-scan.sh', needed by `all-am'.  Stop.
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/tmp/netdata-kickstart-TcKQW2/netdata-v1.19.0-503-g13d7fc4e'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/tmp/netdata-kickstart-TcKQW2/netdata-v1.19.0-503-g13d7fc4e'
make: *** [all] Error 2
 FAILED
 FAILED
 ABORTED  netdata-installer.sh exited with error
```

##### OS / Environment
CentOS release 6.10 (Final)

**Comment by 01CGAT** [comment]
When running "bash <(curl -Ss https://my-netdata.io/kickstart.sh) --dont-wait" for the second time I get:

```
Running in non-interactive mode
yum -y install Judy-devel lz4-devel
Loaded plugins: fastestmirror
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: mirrors.supportex.net
 * centos-sclo-rh: centos.mirror.transip.nl
 * centos-sclo-sclo: centos.mirror.transip.nl
 * extras: centos.mirror.transip.nl
 * remi-safe: mirrors.ukfast.co.uk
 * updates: centos.mirror.transip.nl
No package Judy-devel available.
No package lz4-devel available.
Error: Nothing to do
```

**Comment by Ferroin** [comment]
@01CGAT For CentOS 6 you'll need to enable EPEL (`yum install epel-release`) and the [OKay](https://okay.network/blog-news/rpm-repositories-for-centos-6-and-7.html) repositories for the installation to work.

@joelhans This should be in the install documentation, but seems to have gotten lost somewhere...

**Comment by joelhans** [comment]
@Ferroin I'll create a new issue for us to decide how to best document this. With all this CentOS weirdness (8 has some specifics, too), we should probably create a separate CentOS page with these different instructions.

These EPEL/OKay instructions are actually still in the docs, still in the manual install method, although that's on a different page now: https://docs.netdata.cloud/packaging/installer/methods/manual/

![image](https://user-images.githubusercontent.com/1153921/74449737-399ece80-4e3a-11ea-9b4c-7cfc0defc7a4.png)

Of course, not a user's fault that these instructions are not very discoverable, particularly if they're just trying the kickstart.

**Comment by prologic** [comment]
> @01CGAT For CentOS 6 you'll need to enable EPEL (`yum install epel-release`) and the [OKay](https://okay.network/blog-news/rpm-repositories-for-centos-6-and-7.html) repositories for the installation to work.
> 
> @joelhans This should be in the install documentation, but seems to have gotten lost somewhere...

This is also part of the installer itself with the `install-required-packages.sh` which will go install EPEL / Okay automatically if in non-interactive mode (_otherwise will prompt the user_).

I think the OP has a different issue though which looks suspiciously like a . build failure to me; I could be wrong. Worth spinning up a CentOS 6.x and seeing if we can repro this.

**Comment by 01CGAT** [comment]
After manually installing the Judy-devel and lz4-devel from centos.pkgs.org I now get the following error:

`/tmp/ccADWbbk.ltrans16.ltrans.o: In function `security_initialize_openssl_client.lto_priv.285':
<artificial>:(.text+0x2f52): undefined reference to `TLS_client_method'
/tmp/ccADWbbk.ltrans16.ltrans.o: In function `security_process_accept':
<artificial>:(.text+0x84a2): undefined reference to `SSL_is_init_finished'
/tmp/ccADWbbk.ltrans18.ltrans.o: In function `security_init':
<artificial>:(.text+0x12a1): undefined reference to `OPENSSL_init_ssl'
/tmp/ccADWbbk.ltrans8.ltrans.o: In function `socket_listen_main_static_threaded':
<artificial>:(.text+0x4948): undefined reference to `TLS_server_method'
/tmp/ccADWbbk.ltrans10.ltrans.o: In function `rrdeng_store_metric_init':
<artificial>:(.text+0x80d0): undefined reference to `EVP_MD_CTX_new'
<artificial>:(.text+0x813a): undefined reference to `EVP_MD_CTX_free'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:3537: netdata] Error 1
make[2]: Leaving directory '/tmp/netdata-kickstart-DXye1S/netdata-v1.19.0-513-g246f9e2d'
make[1]: *** [Makefile:4498: all-recursive] Error 1
make[1]: Leaving directory '/tmp/netdata-kickstart-DXye1S/netdata-v1.19.0-513-g246f9e2d'
make: *** [Makefile:2343: all] Error 2`

This are the OpenSSL versions:

`# openssl version
OpenSSL 1.1.1d  10 Sep 2019`

`# yum info openssl-devel
Installed Packages
Name        : openssl-devel
Arch        : x86_64
Version     : 1.0.1e
Release     : 58.el6_10`

**Comment by Ferroin** [comment]
> After manually installing the Judy-devel and lz4-devel from centos.pkgs.org I now get the following error:
> 
> ```
> /tmp/ccADWbbk.ltrans16.ltrans.o: In function `security_initialize_openssl_client.lto_priv.285':
> <artificial>:(.text+0x2f52): undefined reference to `TLS_client_method'
> /tmp/ccADWbbk.ltrans16.ltrans.o: In function `security_process_accept':
> <artificial>:(.text+0x84a2): undefined reference to `SSL_is_init_finished'
> /tmp/ccADWbbk.ltrans18.ltrans.o: In function `security_init':
> <artificial>:(.text+0x12a1): undefined reference to `OPENSSL_init_ssl'
> /tmp/ccADWbbk.ltrans8.ltrans.o: In function `socket_listen_main_static_threaded':
> <artificial>:(.text+0x4948): undefined reference to `TLS_server_method'
> /tmp/ccADWbbk.ltrans10.ltrans.o: In function `rrdeng_store_metric_init':
> <artificial>:(.text+0x80d0): undefined reference to `EVP_MD_CTX_new'
> <artificial>:(.text+0x813a): undefined reference to `EVP_MD_CTX_free'
> collect2: error: ld returned 1 exit status
> make[2]: *** [Makefile:3537: netdata] Error 1
> make[2]: Leaving directory '/tmp/netdata-kickstart-DXye1S/netdata-v1.19.0-513-g246f9e2d'
> make[1]: *** [Makefile:4498: all-recursive] Error 1
> make[1]: Leaving directory '/tmp/netdata-kickstart-DXye1S/netdata-v1.19.0-513-g246f9e2d'
> make: *** [Makefile:2343: all] Error 2
> ```
> 
> This are the OpenSSL versions:
> 
> ```
> # openssl version
> OpenSSL 1.1.1d  10 Sep 2019
> ```
> 
> ```
> # yum info openssl-devel
> Installed Packages
> Name        : openssl-devel
> Arch        : x86_64
> Version     : 1.0.1e
> Release     : 58.el6_10
> ```
> 
> This version of openssl-devel is the latest available version for CentOS 6.

We have known issues at the moment when building on systems with multiple OpenSSL versions installed, that's most likely what you're seeing now.

**Comment by 01CGAT** [comment]
It seems you will need openssl-devel 1.1.x for this to compile, looking at the desired SSL functions on which it errors. So far I'm not aware that openssl-devel 1.1.x is available for CentOS 6.

FWIW: Installing Netdata on CentOS 6 with "bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh)" works fine.

**Comment by Ferroin** [comment]
> It seems you will need openssl-devel 1.1.x for this to compile, looking at the desired SSL functions on which it errors. So far I'm not aware that openssl-devel 1.1.x is available for CentOS 6.
> 
> FWIW: Installing Netdata on CentOS 6 with "bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh)" works fine.

No, it works fine with the version of OpenSSL that's in CentOS 6.x.

The issue here is that when you have both a distribution installed version of OpenSSL and a manually installed local version, our build system ends up using the header files from the local version and the linker flags for the distribution version (which results in build failures if the versions aren't API compatible).

You can in theory work around this by ensuring the build can't see the headers or pkg-config files for the version you don't want to use.

**Comment by amoss** [comment]
@Ferroin I've removed the triage label and assigned you as you seem to understand the issue. I've also added the packaging label which I use as a catch-all for installation related issue, please let me know if we need more specific labeling.

**Comment by prologic** [comment]
As discussed internally this is no longer relevant as the OP did not install the EPEL repo which our `./packaging/installer/install-required-packages.sh` script automatically does not (_in non-interactive mode_) or prompts the user to do so otherwise. This is a pre-req for the source-based installer to work correctly.

**Comment by 01CGAT** [comment]
> As discussed internally this is no longer relevant as the OP did not install the EPEL repo which our `./packaging/installer/install-required-packages.sh` script automatically does not (_in non-interactive mode_) or prompts the user to do so otherwise. This is a pre-req for the source-based installer to work correctly.

I should have mentioned it earlier, but the EPEL and Okay repo's are now installed, after which the same OpenSSL related error's appear due to the difference in installed distribution version and a manually installed local version of OpenSSL, as mentioned by @Ferroin 

PS. the following cosmetic error appears when running the current version of the installer: 
> CentOS Version: 6.10 ...
 > Checking for epel ...
/tmp/netdata-kickstart-upDGJD/install-required-packages.sh: line 1325: [: 6.10: integer expression expected
 > Detected CentOS 6.x ...
 > Checking for Okay ...

**Comment by prologic** [comment]
I can reproduce that error with:

```sh
$ version="6.10"; if [ "$version" -eq 8 ]; then echo "CentOS 8.x"; else echo "NOT CentOS 8.x"; fi
-bash: [: 6.10: integer expression expected
NOT CentOS 8.x
```

**Comment by prologic** [comment]
We should not be doing integer arithmetic or comparison on floats which I'm not even sure any Shell supports :)

```sh
$ version="6.10"; if [[ "${version}" =~ ^8\..*$ ]]; then echo "CentOS 8.x"; else echo "NOT CentOS 8.x"; fi
NOT CentOS 8.x
```

Is the fix => #8339
